bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/events/EncounterEvent.java

```diff
@@ -0,0 +1,29 @@
+package org.bahmni.module.bahmnicore.events;
+
+import org.openmrs.Encounter;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Component
```

Comment:
should not be spring component. 
Use constructor to set internal properties. 

so something like
> new EncounterEvent(bahmniEventType, encounterObject)